common/lib/xmodule/xmodule/library_root_xblock.py

```diff
@@ -71,7 +79,7 @@ def render_children(self, context, fragment, can_reorder=False, can_add=False):
         for child_key in children_to_show:  # pylint: disable=E1101
             child = self.runtime.get_block(child_key)
             child_view_name = StudioEditableModule.get_preview_view_name(child)
-            rendered_child = self.runtime.render_child(child, child_view_name, context)
+            rendered_child = self.render_child(child, child_view_name, context)
```

Comment:
@Kelketek The thing is, when `self.show_children_previews` is False here, we're not really rendering the child at all - we're just calling this class's new `render_child` method which calls runtime's `render_child` method with a magic flag added to the context... And then later in Studio's runtime, we're now watching for that magic flag and if it's set, we wrap the studio header and edit buttons around an empty `Fragment()` and return that. (In the end, it ignores `child_view_name` if `show_preview` is True in the context).

So I'm saying I think the code would be much cleaner if we just acknowledged what the desired end result is right up front, and just use an explicit purpose-made method like `self.runtime.render_placeholder(child)` when `show_children_previews` is True, rather than passing in `context` and `child_view_name` only to have them ultimately ignored by the runtime method called `render_child` that's not actually rendering a child in that case.

I'm not saying this needs to be changed but I think it would be cleaner that way.